src/components/menu/menu-popover.tsx

```diff
@@ -1,14 +1,79 @@
-import { FC, MutableRefObject, useRef } from 'react'
+import { FC, MutableRefObject, useEffect, useRef } from 'react'
 import { useClickOutside } from '../../hooks/use-click-outside'
 import { useMenuContext } from './menu-context'
 import { ElMenuPopover } from './styles'
 
+// close menu when focus move out of menu
+export const menuListFocusOutHandler = (menuList: HTMLDivElement, closeMenu: VoidFunction) => (event) => {
+  if (!menuList.contains((event as FocusEvent).relatedTarget as Node)) {
+    closeMenu()
+  }
+}
+
+// focus to the first menuitems when "arrowDown" pressed while `Menu` open
+export const menuButtonKeyDownHandler = (menuItems: NodeListOf<HTMLElement>) => (event) => {
+  if (event.key === 'ArrowDown') {
+    if (menuItems.length) menuItems[0].focus()
+  }
+}
+
+// handle basic navigation key refer to https://www.w3.org/WAI/ARIA/apg/patterns/menubar
+export const menuItemKeyDownHandler =
+  (menuButton: HTMLButtonElement, menuItems: NodeListOf<HTMLElement>, index: number, closeMenu: VoidFunction) =>
+  (event) => {
+    switch (event.key) {
+      case 'ArrowDown': {
+        const nextItem = menuItems[(index + 1) % menuItems.length]
+        nextItem.focus()
+        break
+      }
+      case 'ArrowUp': {
+        event.preventDefault()
+        const prevItem = menuItems[(index - 1 + menuItems.length) % menuItems.length]
+        prevItem.focus()
+        break
+      }
+      case 'Escape':
+        closeMenu()
+        menuButton.focus()
+        break
+      case 'Enter':
+      case ' ': {
+        const menuItem = menuItems[index]
+        menuItem.click()
+        break
+      }
+    }
+  }
+
 export const MenuPopover: FC = ({ children }) => {
   const { isOpen, closeMenu, getPopoverProps } = useMenuContext()
   const popoverRef = useRef<HTMLDivElement>(null)
 
   useClickOutside(popoverRef, closeMenu)
 
+  useEffect(() => {
+    if (isOpen) {
+      const menuContainer = popoverRef.current?.parentElement
+      const menuItems = menuContainer!.querySelectorAll('[role="menuitem"]') as NodeListOf<HTMLElement>
+      const menuButton = menuContainer!.querySelector('[role="button"][aria-expanded="true"]') as HTMLButtonElement
+      const menuList = menuContainer?.querySelector('[role="menu"]') as HTMLDivElement
+
+      const controller = new AbortController()
+      const { signal } = controller
+
+      menuList.addEventListener('focusout', menuListFocusOutHandler(menuList, closeMenu), { signal })
+      menuButton.addEventListener('keydown', menuButtonKeyDownHandler(menuItems), { signal })
+      menuItems.forEach((menuItem, index) => {
```

Comment:
**question:** instead of needing to attach specific event handler to each menu item, can the menu list have a single handler that catches keydown events that bubble up from the items?

Reply:
I think that is also possible, I'll try it, that is good Idea